shell.nix

```diff
@@ -0,0 +1,18 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+let
+  apple-notes-liberator = pkgs.fetchurl {
+    url = "https://github.com/HamburgChimps/apple-notes-liberator/releases/download/v1.0.4/apple-notes-liberator.jar";
```

Comment:
is there a way to point this `latest` instead of `v1.0.4`? And what does the sha hash do/mean?

Reply:
I am not sure there is a URL that Github will create for latest - would love to be wrong about that! **I will have a look and confirm either way**. 

If there _isn't_ a handy URL for latest versions, I think this file would need to be updated when a new release is made - which might be overheard you don't want.

The SHA is a checksum to ensure that the file isn't tampered with (by tampered I mean someone changed the file, or the person downloading it was man in the middle'd by someone), this hash is always the same as long as an identical JAR is downloaded from Github. 

If the file was ever tampered with, the checksum would change, and Nix would tell the user this. When a release goes up on Github, it's typically stable - meaning the file content for that release will never change.

Reply:
There is a handy URL, but perhaps then it is not so useful in this case because one would have to update the SHA anyway, so perhaps its easier to just update the version number in the url here as well. What do you think?

Handy URL example:

https://github.com/HamburgChimps/jbang-catalog/blob/d1e5826e1f89f91a658054f583763d7285596a91/jbang-catalog.json#L5

Reply:
Ah that is very handy, thank you for providing that.

Yeah, ultimately your point about updating the SHA is correct, you'd want to just update this file on every release. I don't mind doing that if you want me to, it only takes a few seconds to do each time. 

I do find it very convenient to have a nix shell for java projects, and maybe others can benefit from that too, so I see the small amount of co-ordination worth it.

Reply:
This is looking great! Approved! And thank you again!